docs/proposals/scheduling/20221227-node-resource-reservation.md

```diff
@@ -0,0 +1,246 @@
+      ```yaml
+      apiVersion: v1
+      kind: Node
+      metadata:
+        name: fake-node
+        annotations: # specific 5 cores will be calculated, e.g. 0, 1, 2, 3, 4, and then those core will be reserved.
+          node.koordinator.sh/reservation: '{"resources":{"cpu":"5"}}'
+      ```
+   2. just reserve a few explicit cpus, for example 0-3, then the other pods will not use the four cpus 0, 1, 2, 3.
+      ```yaml
+      apiVersion: v1
+      kind: Node
+      metadata:
+        name: fake-node
+        annotations: # the cores 0, 1, 2, 3 will be reserved.
+          node.koordinator.sh/reservation: '{"reservedCPUs":"0-3"}'
+      ```
+   3. both of the previous ways are declared, handled as reserve cpu by `ReservedCPUs`
+      ```yaml
+      apiVersion: v1
+      kind: Node
+      metadata:
+        name: fake-node
+        annotations: # This case will be handled according to the configuration of the reservedCPUs field, the cores 0, 1, 2 and 3 will be reserved and the resources field will be ignored.
+          node.koordinator.sh/reservation: '{"resources":{"cpu":"2"},"reservedCPUs":"0-3"}'
+      ```
+    
+    So either by quantity or by specifying the cpu directly, the exact core that is reserved will be calculated, and then written to nodetopo.annotation["node.koordinator.sh/reservation"]
+
+2. cpu supress:  
+   we should ignore the cpus reserved by node when supress. Here we calculate the resources that BE pod can use, based on the following formula, and then modify the cgroup.
+   ```
+   suppress(BE) := node.Total * SLOPercent - pod(LS).Used - max(system.Used, nodeRserv)
+   ``` 
+   - by quota  
+      parse node.annotation["node.koordinator.sh/reservation"] to get the reserved `resourcelist`, then subtract this resource from the original formula  
+   - by cpuset  
+      read the specific cpu to be reserved on the nodetopo directly, not counting the cpus reserved from the annotation when suppressing  
+      
+
+3. runtimehooks:  
+it will ignored the cpus reserved by `nodetopo.annotations` when create LS pod. 
+   ```
+   cpus(allocatable) = All - pod(LSE).Used - pod(LSR).Used - cpus(reservedByNode)
+   ```  
+  
+<font color=Chocolate>*description:*</font> 
+LS pod use cpus from the shared pool, so here we can ensure that LS pods do not use cpu that is already reserved on node.annotation.
+
+
+#### koord-manager  
+we should update batch-resource on node here, just like `batch-memory`,`batch-cpu`. 
+the implementation of this part depends on the `nodeResource`  controller in koord-manager, 
+so for batch-resource, the `batchResourceFit` plugin can take into account the resources reserved by the node.
+The detailed algorithm is as follows:
+```
+node.alloc = node.alloc - reservedBynode
+node.reserved = node.alloc * reserveRatio
+system.used = max(node.used - pod.used, node.reserved)
```

Comment:
the name of reservXXX maybe misleading

System.used = max(node.used - pod.used, node.anno.reserved)
Pod(Product).Used include LSE/LSR/LS pods now.

Node(Batch).Alloc = Node.Total * thresholdRatio - System.Used - Pod(Product).Used